src/util/source_location.h

```diff
@@ -138,6 +138,16 @@ class source_locationt:public irept
     return get_bool(ID_hide);
   }
 
+  inline bool is_built_in() const
+  {
+    std::string file=get_file().c_str();
+    std::string built_in="<built-in-";//as in "<built-in-additions>";
+    std::string built_in2="<builtin-";//as in "<builtin-architecture-strings>";
+    bool is_built_in=!file.compare(0,built_in.length(),built_in);
+    is_built_in|=!file.compare(0,built_in2.length(),built_in2);
+    return is_built_in;
+  }
```

Comment:
This seems very useful! A few suggestions, though: (1) Don't make it inline, implement this in source_location.cpp. (2) Once done so, make use of has_prefix.